src/Application/Commands/DeleteStalePaymentDetails.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Application\Commands;
+
+use Psr\Log\LoggerInterface;
+use Stripe\StripeClient;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Deletes payment methods from Stripe if:
+ *   1. they (or more specifically, their Customer) are 24h+ old; and
+ *   2. they do not belong to a Person with a password set after mid-April 2023; and
+ *   3. they are a credit/debit card based method; and
+ *   4. they have not been used for any complete donations.
+ */
+class DeleteStalePaymentDetails extends LockingCommand
+{
+    protected static $defaultName = 'matchbot:delete-stale-payment-details';
+
+    public function __construct(
+        private readonly \DateTimeImmutable $initDate,
+        private LoggerInterface $logger,
+        private readonly StripeClient $stripeClient,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription('Deletes unused and likely-non-customer-deletable payment methods from Stripe');
+    }
+
+    protected function doExecute(InputInterface $input, OutputInterface $output): int
+    {
+        $stripePageSize = 100; // Maximum allowed. Iterators page through automatically.
+        $customerCount = 0;
+        $methodsDeleted = 0;
+        $oneDayAgo = $this->initDate
+            ->sub(new \DateInterval('P1D'))
+            ->getTimestamp();
+
+        // Get all Stripe customers without the password set metadata field, who are over 24h old.
+        // The metadata restriction lets us better leave people with passwords since April 2023,
+        // so this `query` covers conditions (1) and (2) from the class doc block.
+        $customers = $this->stripeClient->customers->search([
+            'query' => "created<$oneDayAgo and metadata['hasPasswordSince']:null",
+            'limit' => $stripePageSize,
+        ]);
```

Comment:
is this not going to find the same donors every time the command is run? How will we ever move beyond the first 100 donors?

Reply:
@bdsl See the next line and the auto paging Stripe docs link in my other comment

Reply:
I don't follow. The state of the auto paging iterator is held in PHP memory isn't it? But we need persistent state somewhere to tell us which people have or haven't already had their stale methods detached so we can work on detaching methods from new people every time. Maybe we can discuss tomorrow.

Reply:
@bdsl Right, with you. It's a bit wasteful to keep calling out for the same empty list of attached payment methods for old customers already processed – but if the total list was short enough to complete in an hour and we did this for a few days max. I don't think it's out of the question.

So 2 options I can think of so far:
1. release this as an MVP, then once we've tested it and seen live customers tidied up, add a constraint so that we look at only customers who are also _newer than_ (e.g.) 1 day for potential payment method detachment; or
2. set an additional metadata field on a customer to indicate when we last detached stale payment methods, and update this query to also only look at customers where that is null

Reply:
Yeah, unfortunately the query-able fields are very few. https://stripe.com/docs/search#query-fields-for-customers